packages/kit/types/index.d.ts

```diff
@@ -324,15 +320,28 @@ export interface ServerLoadEvent<
 }
 
 export interface Action<
-	Params extends Partial<Record<string, string>> = Partial<Record<string, string>>
+	Params extends Partial<Record<string, string>> = Partial<Record<string, string>>,
+	OutputData extends Record<string, any> | void = Record<string, any> | void
 > {
-	(event: RequestEvent<Params>): MaybePromise<
-		| { status?: number; errors: Record<string, any>; location?: never }
-		| { status?: never; errors?: never; location: string }
-		| void
-	>;
+	(event: RequestEvent<Params>): MaybePromise<OutputData>;
 }
 
+export type Actions<
+	Params extends Partial<Record<string, string>> = Partial<Record<string, string>>,
+	OutputData extends Record<string, any> | void = Record<string, any> | void
+> = Record<string, Action<Params, OutputData>>;
+
+/**
+ * When calling a form action via fetch, the response will be one of these shapes.
+ */
+export type FormFetchResponse<
```

Comment:
still not sure I understand why this is distinct from `ActionResult`, since errors will (I think?) be serialized as `{ type: 'error', error: any }`